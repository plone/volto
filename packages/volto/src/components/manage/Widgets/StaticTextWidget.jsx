import PropTypes from 'prop-types';

import FormFieldWrapper from '@plone/volto/components/manage/Widgets/FormFieldWrapper';

const StaticTextWidget = (props) => {
  const { id, value } = props;

  return (
    <FormFieldWrapper {...props} className="text" columns={1}>
      <div id={id} className="wrapper">
        <p
          dangerouslySetInnerHTML={{
            __html: value?.data || '',
          }}
        />
      </div>
    </FormFieldWrapper>
  );
};

export default StaticTextWidget;

StaticTextWidget.propTypes = {
  id: PropTypes.string.isRequired,
  value: PropTypes.string,
  minLength: PropTypes.number,
  maxLength: PropTypes.number,
};

StaticTextWidget.defaultProps = {
  value: null,
  minLength: null,
  maxLength: null,
};
