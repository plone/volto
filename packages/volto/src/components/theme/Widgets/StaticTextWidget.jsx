import React from 'react';
import cx from 'classnames';

const StaticTextWidget = ({ value, className }) =>
  value ? (
    <p
      className={cx(className, 'statictext', 'widget')}
      dangerouslySetInnerHTML={{
        __html: value.data,
      }}
    />
  ) : (
    ''
  );

export default StaticTextWidget;
