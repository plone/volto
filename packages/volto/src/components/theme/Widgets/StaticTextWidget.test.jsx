import React from 'react';
import renderer from 'react-test-renderer';
import StaticTextWidget from './StaticTextWidget';

describe('StaticTextWidget', () => {
  it('renders an empty static text view widget component', () => {
    const component = renderer.create(<StaticTextWidget />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a static text view widget component', () => {
    const component = renderer.create(
      <StaticTextWidget
        className="metadata"
        value={{ data: '<b>Foo bar</b>' }}
      />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders a static text view widget component with children', () => {
    const component = renderer.create(
      <StaticTextWidget className="metadata" value={{ data: '<b>Foo bar</b>' }}>
        {(child) => <strong>{child}</strong>}
      </StaticTextWidget>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
