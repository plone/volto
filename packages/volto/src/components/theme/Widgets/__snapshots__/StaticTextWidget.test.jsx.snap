// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticTextWidget renders a static text view widget component 1`] = `
<p
  className="metadata statictext widget"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>Foo bar</b>",
    }
  }
/>
`;

exports[`StaticTextWidget renders a static text view widget component with children 1`] = `
<p
  className="metadata statictext widget"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<b>Foo bar</b>",
    }
  }
/>
`;

exports[`StaticTextWidget renders an empty static text view widget component 1`] = `null`;
